### PR TITLE
add io.github.trigg.discover_overlay

### PR DESCRIPTION
**Pull request by trigg**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*

- x11, ipc, wayland, dri for rendering to screen
- --share=network to connect to localhost Discord & to get user avatar images

- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. ~~If not, I contacted upstream developers about submitting their software to Flathub. **Link:**~~
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. ~~*(If not, explain why.)*~~

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

Original project is https://github.com/trigg/Discover/

Thank you for your consideration, 
Trigg

### CODE REVIEW

File: io.github.trigg.discover_overlay.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+app-id: io.github.trigg.discover_overlay
+runtime: org.gnome.Platform
+runtime-version: '3.38'
```
Comment (hfiguiere) [inline io.github.trigg.discover_overlay.yaml:3]:
```suggestion
runtime-version: '41'
```

this version is deprecated

File: io.github.trigg.discover_overlay.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+sdk: org.gnome.Sdk
+command: discover-overlay
+modules: 
+  - python3-discover-overlay.json
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
```
Comment (hfiguiere) [inline io.github.trigg.discover_overlay.yaml:11]:
```suggestion
  - --socket=fallback-x11
  - --socket=wayland
```

if it supports wayland

File: io.github.trigg.discover_overlay.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+app-id: io.github.trigg.discover_overlay
+runtime: org.gnome.Platform
+runtime-version: '3.38'
+sdk: org.gnome.Sdk
+command: discover-overlay
+modules:
```
Comment (hfiguiere) [inline io.github.trigg.discover_overlay.yaml:6]: modules is usually last.

File: io.github.trigg.discover_overlay.yaml (hunk 1 above)
Comment (hfiguiere) [reply io.github.trigg.discover_overlay.yaml:3]: also does it really need the GNOME platform?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build io.github.trigg.discover_overlay

**Comment by hfiguiere** [comment]
appdata is missing `content_rating` and `releases`

File: io.github.trigg.discover_overlay.yaml (hunk 2 above)
Comment (trigg) [reply io.github.trigg.discover_overlay.yaml:11]: The explanation for fallback led me to believe I wouldn't want it as in some cases the app needs to prefer X11. Am I wrong about that?

File: io.github.trigg.discover_overlay.yaml (hunk 1 above)
Comment (trigg) [reply io.github.trigg.discover_overlay.yaml:3]: It doesn't work with Freedesktop or KDE platforms.

**Comment by trigg** [comment]
Tested on my machines, working as expected.

**Comment by nedrichards** [comment]
bot, build io.github.trigg.discover_overlay

**Pull request closed without merging**